### ISSUE TITLE
Support Log Accumulation

### ISSUE THREAD

**Issue by sgairo** [vmware/vic#6213]
**User Statement:**
As a customer of vic, I don't want performance to suffer at the hand of improved logging. 

**Details:**
"This is a mechanism that allows capture debug/trace level data at the point in time of the call, but defers the actual logging of it. The decision as to what granularity should be logged from the accumulator is deferred until completion of the operation."

**Acceptance Criteria:**
"Success will cause only minimal logging, expected/permitted errors will log a little more, unexpected errors can log significant detail. This allows for very detailed logging during error conditions without impacting normal operation. Requires trigger to dump accumulated trace on panic/process death/OOM kill."

**Comment by hmahmood** [comment]
This should be part of the logging framework epic #3051 . Not sure if this should be an epic of its own though.

**Comment by sgairo** [comment]
I agree that this should fall under the #3051 epic. Though I think the jury is still out on whether or not this shouldn't be an epic. My instinct is that it should be, however that will be made clear after the initial investigation has been done. I have a feeling that many vertically aligned stories will need to fall under this epic.
